src/gt4py/next/iterator/type_inference.py

```diff
@@ -477,6 +477,39 @@ def visit_FunCall(
                 )
                 constraints.add((tup, val))
                 return Val(kind=kind, dtype=elem, size=size)
+            if node.fun.id == "cast_":
```

Comment:
The alternative would be to bring type of type into the typesystem, right?

Reply:
Yes, I discussed the idea with @egparedes a while ago (in the context of the frontend). I believe the conclusion was to wait until we really need this.